crates/common/src/paths.rs

```diff
@@ -36,6 +53,41 @@ pub fn resolve_manifest_file_path(provided_path: impl AsRef<Path>) -> Result<Pat
     }
 }
 
+/// Starting from the current directory, searches upward through
+/// the directory tree for a manifest (that is, a file with the default
+/// manifest name `spin.toml`). If found, the path to the manifest
+/// is returned, with a boolean flag indicating if the found path was
+/// the default (i.e. `spin.toml` in the current directory).
+/// If no matching file is found, the function returns None.
+///
+/// The search is abandoned if it reaches the root directory, or the
+/// root of a Git repository, without finding a 'spin.toml'.
+pub fn search_upwards_for_manifest() -> Option<(PathBuf, usize)> {
+    let candidate = PathBuf::from(DEFAULT_MANIFEST_FILE);
+
+    if candidate.is_file() {
+        return Some((candidate, 0));
+    }
+
+    for distance in 1..20 {
+        let inferred_dir = PathBuf::from("../".repeat(distance));
```

Comment:
FYI: https://doc.rust-lang.org/std/path/struct.Path.html#method.ancestors

e.g. `for candidate in Path::new(".").ancestors().take(20)`

Reply:
I don't think this does what I think you're suggesting.  In my tests `Path::new(".").ancestors()` produced `.` then the empty string, then stops.  Are you seeing something different?  Maybe depends on OS?

Reply:
Oh sure enough. I glanced at the example in its docs, saw `..` and mindlessly thought "that looks about right!"...